shared/chat/conversations-list/container.js

```diff
@@ -1,138 +1,37 @@
-    const hasUnread = !!unreadCount
-    const subColor = isSelected ? globalColors.black_40 : hasUnread ? globalColors.white : globalColors.blue3_40
-    const showBold = !isSelected && hasUnread
-    const backgroundColor = isSelected ? globalColors.white : hasUnread ? globalColors.darkBlue : globalColors.darkBlue4
-    const usernameColor = isSelected ? globalColors.black_75 : hasUnread ? globalColors.white : globalColors.blue3_60
-
-    return {
-      backgroundColor,
-      hasUnread,
-      participantNeedToRekey,
-      showBold,
-      subColor,
-      usernameColor,
-      youNeedToRekey,
-    }
-  }
-
-  render () {
-    const pendingRows = this.props.pending.map((users, conversationIDKey) => {
-      const unreadCount = 0
-      const participants = participantFilter(users, this.props.you)
-      const isSelected = this.props.selectedConversation === conversationIDKey
-      const isMuted = false
-      const rekeyInfo = null
-      const timestamp = formatTimeForConversationList(Date.now(), this.props.nowOverride)
-      const snippet = ''
-      const onSelectConversation = this.props.onSelectConversation
-
-      return {
-        conversationIDKey,
-        isMuted,
-        isSelected,
-        onSelectConversation,
-        participants,
-        rekeyInfo,
-        snippet,
-        timestamp,
-        unreadCount,
-      }
-    }).toList()
+const createImmutableEqualSelector = createSelectorCreator(defaultMemoize, I.is)
 
-    const realRows = this.props.inbox.map(conversation => {
-      const conversationIDKey: ConversationIDKey = conversation.get('conversationIDKey')
-      const unreadCount = this.props.conversationUnreadCounts.get(conversationIDKey)
-      const participants = participantFilter(conversation.get('participants'), this.props.you)
-      const isSelected = this.props.selectedConversation === conversationIDKey
-      const isMuted = conversation.get('muted')
-      const rekeyInfo = this.props.selectedConversation && this.props.rekeyInfos.get(conversationIDKey)
-      const timestamp = formatTimeForConversationList(conversation.get('time'), this.props.nowOverride)
-      const snippet = conversation.get('snippet')
-      const onSelectConversation = this.props.onSelectConversation
-
-      return {
-        conversationIDKey,
-        isMuted,
-        isSelected,
-        onSelectConversation,
-        participants,
-        rekeyInfo,
-        snippet,
-        timestamp,
-        unreadCount,
-      }
-    })
-
-    const rows = pendingRows.concat(realRows).map(props => ({
-      ...props,
-      ...this._derivedProps(props.rekeyInfo, props.unreadCount, props.isSelected),
-    }))
-
-    return <ConversationList
-      children={this.props.children}
-      onNewChat={this.props.onNewChat}
-      onSelectConversation={this.props.onSelectConversation}
-      rows={rows}
-    />
+const filteredInbox = createImmutableEqualSelector(
```

Comment:
I wanted to keep the store the same (short term) so this is a reselect so we don't do any work if the filtered inbox is ultimately the same